File: js/wordcount.js
import readline from "readline"
const reader = readline.createInterface({ input: process.stdin })
const counts = new Map()

reader
  .on("line", line => {
    for (let [word] of line.toLowerCase().matchAll(/[\p{L}’']+/gu)) {
      counts.set(word, (counts.get(word) ?? 0) + 1)
    }
  })
  .on("close", () => {
    for (let [word, count] of [...counts.entries()].sort()) {
      console.log(`${word} ${count}`)
    }
  })

File: lua/wordcount.lua
counts = {}
for line in io.lines() do
  line:lower():gsub('[a-z\']+', function(word)
    counts[word] = (counts[word] or 0) + 1
  end)
end

report = {}
for word, count in pairs(counts) do
  table.insert(report, string.format('%s %d', word, count))
end
table.sort(report)
for _, line in ipairs(report) do
  print(line)
end
